### PR TITLE
using structopt and adding Route stops

### PR DESCRIPTION
**Pull request by prhod**
This PR is a mid progress update. There is still some work to do on the extraction of the RoutePoints and their categorization.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -204,6 +210,26 @@ fn osm_route_to_shape(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmO
...
+    osm_relation.refs
+        .iter()
+        .filter(|refe| stop_roles.contains(&refe.role.as_str()))
+        .filter_map(|refe| match refe.member {
+            osmpbfreader::OsmId::Node(obj_id) => Some(format!("StopPoint:Node:{}", obj_id.0)),
+            osmpbfreader::OsmId::Way(obj_id) => Some(format!("StopPoint:Way:{}", obj_id.0)),
+            osmpbfreader::OsmId::Relation(obj_id) => {
+                Some(format!("StopPoint:Relation:{}", obj_id.0))
+            }
+        })
```
Comment (TeXitoi) [inline src/lib.rs]: you can just map here, and an `use osmpbfreader::OsmId` can clean a bit the noise
```rust
use osmpbfreader::OsmId;
...
.map(|refe| match refe.member {
    OsmId::Node(obj_id) => format!("StopPoint:Node:{}", obj_id.0),
    OsmId::Way(obj_id) => format!("StopPoint:Way:{}", obj_id.0),
    OsmId::Relation(obj_id) => format!("StopPoint:Relation:{}", obj_id.0),
})
```

File: src/lib.rs [hunk 2]
```diff
@@ -235,6 +262,7 @@ fn osm_obj_to_line(obj_map: &BTreeMap<osmpbfreader::OsmId, osmpbfreader::OsmObj>
             id: format!("Line:Relation:{}", rel.id.0),
             name: rel.tags.get("name").cloned().unwrap_or("".to_string()),
             code: rel.tags.get("ref").cloned().unwrap_or("".to_string()),
+            colour: rel.tags.get("colour").cloned().unwrap_or("".to_string()),
```
Comment (TeXitoi) [inline src/lib.rs]:
```rust
color: rel.tags.get("color").cloned().unwrap_or_default(),
```

File: src/main.rs [hunk 3]
```diff
@@ -30,50 +30,44 @@
...
 
-#[derive(RustcDecodable, Debug)]
+#[derive(StructOpt)]
 struct Args {
-    flag_input: String,
-    flag_connection_string: String,
-    flag_import_stop_points: bool,
-    flag_import_routes: bool,
-}
+    //#[structopt(long = "help", short = "h", help = "Show this message")]
```
Comment (TeXitoi) [inline src/main.rs]: You can remove that.

File: src/main.rs [hunk 4]
```diff
@@ -30,50 +30,44 @@
...
+    #[structopt(long = "input", short = "i", help = "OSM PBF file")]
+    input: String,
 
-static USAGE: &'static str = "
-Usage:
-    osmtc2mongo --help
-    osmtc2mongo --input=<file> [--connection-string=<connection-string>] [--import-stop-points] [--import-routes]
+    #[structopt(long = "import-stop-points-only", short = "s",
+                help = "Import only stop_points (dafault is a full extraction)")]
```
Comment (TeXitoi) [inline src/main.rs]: dafault -> default

File: src/lib.rs [hunk 5]
```diff
@@ -75,6 +74,7 @@ pub struct Line {
     pub id: String,
     pub name: String,
     pub code: String,
+    pub colour: String,
```
Comment (prhod) [inline src/lib.rs]: colour is the osm name : http://wiki.openstreetmap.org/wiki/Proposed_features/Public_Transport#Route_direction.2Fvariant

Comment (TeXitoi) [inline src/lib.rs]: OK

**Pull request merged**